test/SPI/warn_on_ineffective_spi_import.swift

```diff
@@ -0,0 +1,19 @@
+/// Test the warning on an SPI import of the public interface of a module.
+
+// RUN: %empty-directory(%t)
+
+/// Compile the SPI lib.
+// RUN: %target-swift-frontend -enable-experimental-prespecialization -emit-module %S/Inputs/spi_helper.swift -module-name SPIHelper -emit-module-path %t/SPIHelper.swiftmodule -emit-module-interface-path %t/SPIHelper.swiftinterface -emit-private-module-interface-path %t/SPIHelper.private.swiftinterface -enable-library-evolution -swift-version 5 -parse-as-library
+
+/// Reading from swiftmodule, no warning.
+// RUN: %target-swift-frontend -typecheck %s -I %t
+
+/// Reading from .private.swiftinterface, no warning.
+// RUN: rm %t/SPIHelper.swiftmodule
+// RUN: %target-swift-frontend -typecheck %s -I %t
+
+/// Reading from the public .swiftinterface should produce the warning.
+// RUN: rm %t/SPIHelper.private.swiftinterface
+// RUN: %target-typecheck-verify-swift -I %t
+
+@_spi(SPIHelper) import SPIHelper // expected-warning {{importation of 'SPIHelper' has an ineffective '@_spi' attribute; it was built from the public interface at}}
```

Comment:
This message is a bit confusing -- can it say something along the lines of `import of 'SPIHelper' will not include SPI symbols, since it was built from the public interface at ...`

And could you add a note with a Fix-It to remove the @_spi attribute?

Reply:
Thank you for the suggestion, I updated the diagnostic message.

For now I prefer to avoid having a fix-it here because removing the attribute is not the solution to where I expect to see this error. The likely solution is probably to change the framework search paths or something like that.